Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.FileDialog.cs

```diff
@@ -0,0 +1,282 @@
+
+namespace System.Windows.Forms
+{
+    // <summary>
+    // Displays a dialog box from which the user can select a file.
+    // </summary>
+    public abstract class FileDialog : CommonDialog
+    {
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box automatically adds an extension to a file name if the user omits the extension.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box adds an extension to a file name if the user omits the extension; otherwise, false. The default value is true.
+        // </returns>
+        // public bool AddExtension {get; set;}
+
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box displays a warning if the user specifies a file name that does not exist.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box displays a warning if the user specifies a file name that does not exist; otherwise, false. The default value is false.
+        // </returns>
+        // [DefaultValue(false)]
+        // public virtual bool CheckFileExists { get; set; }
+
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box displays a warning if the user specifies a path that does not exist.
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box displays a warning when the user specifies a path that does not exist; otherwise, false. The default value is true.
+        // </returns>
+        // public bool CheckPathExists {get; set;}
+
+        // <summary>
+        // Gets or sets the default file name extension.
+        // </summary>
+        // 
+        // <returns>
+        // The default file name extension. The returned string does not include the period. The default value is an empty string ("").
+        // </returns>
+        // public string DefaultExt {get; set;}
+
+        // <summary>
+        // Gets or sets a value indicating whether the dialog box returns the location of the file referenced by the shortcut or whether it returns the location of the shortcut (.lnk).
+        // </summary>
+        // 
+        // <returns>
+        // true if the dialog box returns the location of the file referenced by the shortcut; otherwise, false. The default value is true.
+        // </returns>
+        // public bool DereferenceLinks {get; set;}
+
+        // <summary>
+        // Gets or sets a string containing the file name selected in the file dialog box.
+        // </summary>
+        // 
+        // <returns>
+        // The file name selected in the file dialog box. The default value is an empty string ("").
+        // </returns>
+        //<PermissionSet><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        // public string FileName {get; set;}
```

Comment:
While looking up the Reference Source for `FileNames` it appears that this is guaranteed to be non-null (will be empty if no value). Consider adding a postreq here.